utils/pre-swap-checker/contracts/PreSwapChecker.sol

```diff
@@ -147,6 +149,64 @@ contract PreSwapChecker {
     return (errorCount, errors);
   }
 
+  /**
+    * @notice If order is going through wrapper to swap
+    * @param order Types.Order
+    * @param fromAddress address
+    * @param wrapper address
+    * @return uint256 errorCount if any
+    * @return bytes32[] memory array of error messages
+    */
+  function checkSwapWrapper(
+    Types.Order calldata order,
+    address fromAddress,
+    address payable wrapper
+    ) external view returns (uint256, bytes32[] memory ) {
+    IWETH wethContract =  Wrapper(wrapper).wethContract();
+    address swap = order.signature.validator;
+    // max size of the number of errors that could exist
+    bytes32[] memory errors = new bytes32[](4);
+    uint256 errorCount;
+
+    if (order.sender.wallet != fromAddress) {
+      errors[errorCount] = "MSG_SENDER_MUST_BE_ORDER_SENDER";
+      errorCount++;
+    }
+
+    // ensure that sender has approved wrapper contract on swap
+    if (!ISwap(swap).senderAuthorizations(order.sender.wallet, wrapper)) {
+      errors[errorCount] = "WRAPPER_UNAUTHORIZED";
+      errorCount++;
+    }
+
+    // signature must be filled in order using Wrapper
+    if (order.signature.v == 0) {
+      errors[errorCount] = "SIGNATURE_MUST_BE_SENT";
+      errorCount++;
+    }
+
+
+    // if sender has WETH token, ensure sufficient ETH balance
+    if (order.sender.token == address(wethContract)) {
+      if (address(order.sender.wallet).balance < order.sender.amount) {
+        errors[errorCount] = "SENDER_INSUFFICIENT_ETH";
+        errorCount++;
+      }
+    }
+
+    // ensure that sender wallet if receiving weth has approve it to
+    // transfer weth and deliver eth to the sender
+
+    if (order.signer.token == address(wethContract)) {
+      uint256 allowance = wethContract.allowance(order.sender.wallet, wrapper);
+      if (allowance < order.signer.amount) {
+        errors[errorCount] = "LOW_SENDER_ALLOWANCE_ON WRAPPER";
+        errorCount++;
+      }
+    }
+
+    return (errorCount, errors);
```

Comment:
It would be nice if calling `checkSwapWrapper` performed all of the necessary `swap` checks too - so you dont have to call multiple contracts. But I think that would include copying the contents of an array into another. Thoughts?

Reply:
given it's a call i may just doing the copying all the contents from one to the other. thanks!

Reply:
Mechanism for this to work:

checkSwapSwap would be called first but with some modification. It would need access to the WETH contract address and dictate it was using the Wrapper as the balance check with the PreSwapChecker.checkSwapSwap doesn't necessarily hold for the PreSwapChecker.checkWrapperSwap. The reason is that the balance of WETH may be lower than the balance needed because the sender will be depositing ETH into the WETH contract to get a WETH balance. 

From there checkWrapperSwap can use the errorCount to determine how many slots to copy from the checkSwapErrors array into the checkWrapperSwap error array.